doc/manual/compiler.rst

```diff
@@ -54,74 +76,125 @@ The Python types correspond to ARTIQ type annotations as follows:
+
+Available built-in functions 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
+
+ARTIQ makes various useful built-in and mathematical functions from Python, NumPy, and (if present) SciPy available in kernel code. They are not guaranteed to be perfectly equivalent to their host equivalents (for example, ``numpy.rint`` normally rounds-to-even, but in kernel code rounds toward zero) but their behavior should basically predictable. 
+
+
+.. list-table::
+    :header-rows: 1
+    
+    +   * Reference 
+        * Functions 
+    +   * `Python built-ins <https://docs.python.org/3/library/functions.html>`_
+        *   - ``len()``, ``round()``, ``abs()``, ``min()``, ``max()``
+            - ``print()`` (for specifics see warning below)
+            - all basic type conversions (``int()``, ``float()`` etc.)  
+    +   * `NumPy mathematic utilities <https://numpy.org/doc/stable/reference/routines.math.html>`_
+        *   - ``sqrt()``, ``cbrt``` 
+            - ``fabs()``, ``fmax()``, ``fmin()``
+            - ``floor()``, ``ceil()``, ``trunc()``, ``rint()``
+    +   * `NumPy exponents and logarithms <https://numpy.org/doc/stable/reference/routines.math.html#exponents-and-logarithms>`_
+        *   - ``exp()``, ``exp2()``, ``expm1()``
+            - ``log()``, ``log2()``, ``log10()``
+    +   * `NumPy trigonometric and hyperbolic functions <https://numpy.org/doc/stable/reference/routines.math.html#trigonometric-functions>`_
+        *   - ``sin()``, ``cos()``, ``tan()``, 
+            - ``arcsin()``, ``arccos()``, ``arctan()`` 
+            - ``sinh()``, ``cosh()``, ``tanh()`` 
+            - ``arcsinh()``, ``arccosh()``, ``arctanh()`` 
+            - ``hypot()``, ``arctan2()``
+    +   * `NumPy floating point routines <https://numpy.org/doc/stable/reference/routines.math.html#floating-point-routines>`_
+        *   - ``copysign``, ``nextafter()``
+    +   * `SciPy special functions <https://docs.scipy.org/doc/scipy/reference/special.html>`_ 
+        *   - ``erf()``, ``erfc()``
+            - ``gamma()``, ``gammaln()``
+            - ``j0()``, ``j1()``, ``y0()``, ``y1()``
+
+Basic NumPy array handling (``np.array()``, ``numpy.transpose()``, ``@``, element-wise operation etc.) is also available. NumPy functions are implicitly broadcast when applied to arrays. 
+
+.. warning:: 
+    A kernel ``print`` call normally specifically prints to the host machine, either the terminal of ``artiq_run`` or the dashboard log when using ``artiq_dashboard``. This makes it effectively an RPC, with some of the corresponding limitations. In subkernels and whenever compiled through ``artiq_compile``, where RPCs are not supported, it is instead considered a print to the local UART log.
```

Comment:
>  whenever compiled through ``artiq_compile``

Compiled as an idle or startup kernel with ``artiq_compile``?

> local UART log.

UART only on satellites, master/standalone log is also accessible by network (coremgmt), just won't be displayed on the terminal (as there's no terminal to connect to)

Reply:
Oops, I corrected this and then somehow must have un-corrected it while fistfighting rST table formatting

Reply:
Also, `artiq_compile` doesn't allow RPCs as a matter of principle, and it isn't technically limited to idle or startup kernels, nor does it know that it's working on an idle or startup kernel. So I'd stick with the broader statement, it seems more accurate